sherpa/plot/__init__.py

```diff
@@ -2206,31 +2486,54 @@ class DelchiPlot(ModelPlot):
     plot_prefs = basicbackend.get_resid_plot_defaults()
     "The preferences for the plot."
 
-    def _calc_delchi(self, ylist, staterr):
-        return (ylist[0] - ylist[1]) / staterr
-
-    def prepare(self, data, model, stat):
-        (self.x, y, staterr, self.xerr,
-         self.xlabel, self.ylabel) = data.to_plot(model)
+    def _calc_y(self,
+                data: Data1D,
+                stat: Stat,
+                ylist: tuple[np.ndarray, np.ndarray],
+                staterr: Optional[np.ndarray]) -> None:
+        """Define the self.y and self.yerr fields"""
 
         if staterr is None:
             if stat.name in _stats_noerr:
                 raise StatErr('badstat', "DelchiPlot", stat.name)
             staterr = data.get_yerr(True, stat.calc_staterror)
 
-        self.y = self._calc_delchi(y, staterr)
-        self.yerr = staterr / staterr
+        self.y = (ylist[0] - ylist[1]) / staterr
+        self.yerr = np.ones_like(self.y)
+
+    def _change_ylabel(self) -> None:
         self.ylabel = 'Sigma'
+
+    def _title(self, data: Data1D) -> None:
```

Comment:
So, there's a lot of things that we could do to address this, but do do it generically would just slow down getting something useful to the user. I am getting a bit tired of looking at an area of Sherpa, deciding to rework it, and then getting bogged down in trying to work out how to pay off the technical debt. So I'm trying to focus on user-facing changes since we don't have much time to the next release!

Reply:
That's fine. We're not making it worse; you're just following the patterns that that were established long ago. In fact, I think it's no me to re-work this, I've just not done that yet. So, agreed, let's get this to the user and worry about technical dept some other time.